src/sentry/api/event_search.py

```diff
@@ -539,9 +539,14 @@ def _handle_basic_filter(self, search_key, operator, search_value):
 
         return SearchFilter(search_key, operator, search_value)
 
-    def _handle_numeric_filter(self, search_key, operator, search_value):
+    def _handle_numeric_filter(self, negated, search_key, operator, search_value):
```

Comment:
I tried to incorporate feedback to modify the operator in the caller and avoid having a `negated` param here but I found making the change more difficult than this. Here are a few reasons why I think I'd like to keep this approach:

1) This method has logic to fallback to a text filter (e.g. to handle `!random_field:>500`, and it relies on `negated` at line 565
2) Moving the logic up requires duplicating the `is_numeric_key` check because that changes how we modify the operators and stuff, and that would affect the other filters that use this method like boolean

I think this way we reduce the amount of knowledge required to understand what both methods are doing

Reply:
I take this comment back, I tried it more and got it working by massaging the data in `visit_numeric_field` as suggested. When a numeric field is picked up I check if the key is numeric and if it's not I do the fallback work to use it as a text field